src/arc_utilities/catches.py

```diff
@@ -0,0 +1,65 @@
+import signal
+from typing import Callable, Optional
+
+
+def catch_timeout(seconds: int, func: Callable, *args, **kwargs):
+    def _handle_timeout(signum, frame):
+        raise TimeoutError()
+
+    try:
+        signal.signal(signal.SIGALRM, _handle_timeout)
+        signal.alarm(seconds)
+        try:
+            result = func(*args, **kwargs)
+            return result, False
+        finally:
+            signal.alarm(0)
+    except TimeoutError:
+        print("Caught timeout!")
+        return None, True
+
+
+def retry_on_timeout(t: int, on_timeout: Optional[Callable], f: Callable, *args, **kwargs):
```

Comment:
Why is t an int? can't we specify 1.5 seconds and so on? Same comment with seconds above. Thoughts on renaming `f` to `generator` to be more explicit?